### ISSUE TITLE
'setup_plugin_with_filter_data' should probably not run on after_setup_theme

### ISSUE THREAD

**Issue by wpexplorer** [awesomemotive/one-click-demo-import#145]
We are basically creating a new instance of the demo importer on every page load (backend or front-end) and saving import data to memory when not needed.

In reality we shouldn't need to create an instance prior to the user clicking the import button. But at least don't start anything until you are actually on the import dashboard screen.

In my case I host an endpoint on a multisite that loops through all sub-sites and auto generates the array as json which I can pass on to the pt-ocdi/import_files filter. And while I can set a transient so it's not constantly pulling and parsing the data from the URL it still doesn't make sense for pt-ocdi/import_files to be triggered on every site load by default.

For anyone wondering this is the code I use:

```
function wpex_get_demos_list() {

	$demos = get_transient( 'wpex_get_demos_list' );

	if ( ! $demos ) {

		$json_url = 'URL GOES HERE';

		$response = wp_remote_get( $json_url );

		if ( is_wp_error( $response ) || ( wp_remote_retrieve_response_code( $response ) != 200 ) ) {
			return array();
		}

		$body = wp_remote_retrieve_body( $response );

		if ( ! $body ) {
			return array();
		}

		$demos = json_decode( $body );


		set_transient( 'wpex_get_demos_list', $demos, 48 * HOUR_IN_SECONDS );

	}

	return apply_filters( 'wpex_get_demos_list', $demos );

}
add_filter( 'pt-ocdi/import_files', 'wpex_get_demos_list' );
```

Just like we shouldn't load scripts when not needed we shouldn't be running PHP code when not needed ;)

Thanks!

**Comment by capuderg** [comment]
Hi,

thanks for reporting this issue.

I agree, this should be improved and run only on the actual admin import page.

We are more then happy to accept pull-requests for such improvements :)

Take care!

**Comment by wpexplorer** [comment]
Sounds good. I'll setup a fork so I can send over pull requests. I wasn't sure if you guys preferred to make the edits yourself and if you accepted pull requests. I'm more then happy to mess around with the code and provide pull requests.

By the way @capuderg - I am always looking to hire for freelance work if you are available and also willing to pay for updates to products we will use such as this one.

**Comment by capuderg** [comment]
Awesome! We are always happy to merge a good PR, so go ahead and create one for this issue.

**Comment by capuderg** [comment]
Hi @wpexplorer,

sorry for the late reply.

I've partially improved this issue, by not executing the code in this hook, if a frontend page is loaded.

We'll come back to this issue in the near future and resolve it properly. I just wanted to add this little improvement to the next plugin update which will include mostly bug fixes.

I've tested a few different hooks and tried a few changes, but something always broke on some part of the plugin. I tried hooks that execure later in the WP cycle with the `get_current_screen` to try and run this code only on the plugin page, but then the predefined imports didn't show up or the WP CLI didn't work...

The plugin functionality and its flow needs to be looked at in its entirety to fix this properly.

Take care!
